core/inspector.rs

```diff
@@ -727,11 +760,62 @@ impl Stream for InspectorSession {
   }
 }
 
+/// TODO:
+pub struct LocalInspectorSessionRaw {
+  v8_session_tx: UnboundedSender<String>,
+  v8_session_rx: tokio::sync::Mutex<UnboundedReceiver<InspectorMsg>>,
+}
+
+impl GarbageCollected for LocalInspectorSessionRaw {}
+
+impl LocalInspectorSessionRaw {
+  pub fn new(
+    v8_session_tx: UnboundedSender<String>,
+    v8_session_rx: UnboundedReceiver<InspectorMsg>,
+  ) -> Self {
+    Self {
+      v8_session_tx,
+      v8_session_rx: tokio::sync::Mutex::new(v8_session_rx),
+    }
+  }
+
+  // TODO(bartlomieju): should this return a result or anything else?
+  pub fn post_message<T: serde::Serialize>(
+    &self,
+    id: i32,
+    method: &str,
+    params: Option<T>,
+  ) {
+    let msg = json!({
+      "id": id,
+      "method": method,
+      "params": params
+    });
+    let stringified_msg = serde_json::to_string(&msg).unwrap();
+    self.v8_session_tx.unbounded_send(stringified_msg).unwrap();
+  }
+
+  pub async fn receive_from_v8_session(&self) -> Option<InspectorMsg> {
+    let mut v8_session_rx = self.v8_session_rx.lock().await;
+    (*v8_session_rx).next().await
+  }
+
+  pub fn disconnect(&self) {
+    // TODO(bartlomieju): this should at least have a cancel handle and cancel pending
+    // `self.receive_from_v8_session()` calls
+    eprintln!("LocalInspectorSessionRaw::disconnect not implemented");
+  }
+}
+
+impl Drop for LocalInspectorSessionRaw {
+  fn drop(&mut self) {
+    eprintln!("Dropping LocalInspectorSessionRaw");
+  }
+}
 /// A local inspector session that can be used to send and receive protocol messages directly on
 /// the same thread as an isolate.
 pub struct LocalInspectorSession {
```

Comment:
This struct should probably be let go from `deno_core`, but I'm still weighing it. `node:inspector` needs low level access to handle notifications/message response it self, but REPL, Jupyter and HMR runner rely on high-level API to await responses when posting.